react/components/Onboarding/components/Channels/components/CreateChannel/index.js

```diff
@@ -114,52 +141,68 @@ class CreateChannel extends React.Component {
   };
 
   render() {
-    const hasText = this.state.channelName;
-    const username = currentUserService().username;
+    const { data: { loading, me } } = this.props;
+
+    if (loading) {
+      return null;
+    }
+
+    const { name } = me;
+    const hasText = this.state.channelTitle;
 
     return (
       <CreateChannelWrapper>
         <TransitionGroup style={{display: "block", position: "relative"}}>
           <AnimatedCTAText
             key={
-              `onboarding-create-channel-cta-step-${this.state.channelNameStep}`
+              `onboarding-create-channel-cta-step-${this.state.channelTitleStep}`
             }
-            positionAbsoluteDuringTransition={this.state.channelNameStep == 1}
+            positionAbsoluteDuringTransition={this.state.channelTitleStep == 1}
           >
-            { this.ctaTextForChannelNameStep() }
+            {this.ctaTextForChannelTitleStep()}
           </AnimatedCTAText>
         </TransitionGroup>
         <div>
-          <ChannelNameInput
+          <ChannelTitleInput
             autoFocus={true}
-            value={this.state.channelName}
-            onChange={this.handleChannelNameChange}/>
+            disabled={this.state.saving}
+            onChange={this.handleChannelTitleChange}
+            value={this.state.channelTitle}/>
         </div>
-        <FadingBlockWrapper hasText={ hasText }>
+        <FadingBlockWrapper hasText={hasText}>
           <Block
             type={Types.CHANNEL}
             blockData={{
               length: 0,
-              title: this.state.channelName,
+              title: this.state.channelTitle,
               updatedAtAgo: "0 minutes ago",
-              username: username,
+              username: name,
               visibility: "private"
             }}
           />
         </FadingBlockWrapper>
         <FadingCTAButton
-          hasText={ hasText }
-          onClick={()=>{console.log("forward!")}}
+          disabled={this.state.saving}
+          hasText={hasText}
+          onClick={this.handleSubmit}
         >
-          Create Channel
+          { this.state.saving ? 'Saving…' : 'Create Channel' }
         </FadingCTAButton>
       </CreateChannelWrapper>
     );
   }
 };
 
-CreateChannel.propTypes = {};
+CreateChannel.propTypes = {
+  createChannel: PropTypes.func.isRequired,
+  data: PropTypes.shape({
+    loading: PropTypes.bool.isRequired,
```

Comment:
To easily proptype out the data shape you can use fragments that are used in your queries, like so: https://github.com/aredotna/ervell/blob/fae48e5f288d77251acd688531f700d150a3d5ba/react/components/ProfileMetadata/index.js#L19